src/OpenDiffix.Core/Aggregator.fs

```diff
@@ -50,9 +50,24 @@ let private increaseContribution valueIncrease aidValue (aidMap: Dictionary<AidH
 
   aidMap.[aidHash] <- updatedContribution
 
-type private RequestedOutput =
-  | Value
-  | Noise
+let private mergeOutliers outliers (contributions: Anonymizer.ContributionsState array) =
```

Comment:
I admit I'm finding this part hard to follow - following what is what, and where do we take the data from and which data we mutate... The problem is in the combination of this function here with the `>>` and `getState` construct used in the `Final` members - I have to jump around the file (and `Anonymizer.fs` to follow the data around.

I have however no alternative off-hand. I wish there was a way to code it so it reads more top-to-bottom like a recipe - "take this out of that and put into this other thing".

Maybe an idea would be to duplicate `mergeOutliers` as members of the Aggregator classes first, make them contain all the logic and glue code involved and only then try to abstract out common stuff?

Reply:
Sorry the code seems obtuse, I struggled with it for a while and couldn't find a more elegant version.
This part is common to count and sum aggregators, I don't see how it would help if we duplicate it.
`getState` has to be called twice each time, so we still need to separate it from this part.

Reply:
I can't see the improvement I suggested yesterday anymore, you're right.